libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/IrOperationsBuilder.kt

```diff
@@ -594,7 +608,12 @@ internal class IrOperationsBuilder(
        *   }
        * }
        */
-      usedFields.putType(first.type.rawType().name)
+      usedCoordinates.putType(first.type.rawType().name)
+
+      // When a field with arguments is selected, its parent type is referenced in the compiled selections
+      if (first.definitionHasArguments) {
+        usedCoordinates.putType(first.parentType)
```

Comment:
Could you add all the arguments here instead of calling `UsedCoordinates.putAllArguments` at the end of the process? You'll probably have to add `arguments` to the `CollectedField` but it's colocating everything.